IrisHub/SpaceBar
# Better dependency management

### Description of Changes

- Move many dependencies that were incorrectly included as part of runtime `dependencies` back to `devDependencies` across Spacebar client + server
- Deleted unimported code that imported unused deps such as react/cannon
- Reduced Heroku slug size from `840mb` to `163.5mb` with the following optimizations: 
     -- Introduced a .`slugignore` to exclude node_modules and other unnecessary files
     -- Generating a fresh install of dependencies before building sourcemaps in the Spacebar deployment script 
     --  Excluding dirs such as `node_modules` in a .`slug-post-clean` file after using a custom buildpack that deletes assets used to generate a production build but not required as part of the slug
- Bonus -- Add Jotai debugging libraries for use in dev

### Link to Issue
Link the appropriate GitHub Issue for this PR. You can link an issue by adding `closes #issue-number` to the PR message body.

### Testing Procedure
yarn start & yarn build locally run as expected with no errors for client & server
./spacebar deploy -s client & ./spacebar deploy -s server run as expected & deploy successfully
